plugins/aea-ledger-hwi/aea_ledger_hwi/hwi.py

```diff
@@ -0,0 +1,338 @@
+        """Simplify a list with process value."""
+        return [callable_name(el) for el in li]
+
+    @classmethod
+    def _valid_key(cls, key: Any) -> str:
+        """Check validity of key."""
+        if isinstance(key, str):
+            return key
+        raise ValueError("Key must be string.")  # pragma: nocover
+
+    @classmethod
+    def to_dict(cls, attr_dict: Union[AttributeDict, TxReceipt, TxData]) -> JSONLike:
+        """Simplify to dict."""
+        if not isinstance(attr_dict, AttributeDict):
+            raise ValueError("No AttributeDict provided.")  # pragma: nocover
+        result = {
+            cls._valid_key(key): cls._remove_hexbytes(value)
+            for key, value in attr_dict.items()
+        }
+        return result
+
+    @classmethod
+    def from_dict(cls, di: JSONLike) -> AttributeDict:
+        """Get back attribute dict."""
+        if not isinstance(di, dict):
+            raise ValueError("No dict provided.")  # pragma: nocover
+        processed_dict = {
+            cls._valid_key(key): cls._add_hexbytes(value) for key, value in di.items()
+        }
+        return AttributeDict(processed_dict)
+
+
+class EthereumHWICrypto(Crypto[HWIAccount]):
+    """Class wrapping the Account Generation from Ethereum ledger."""
+
+    identifier = _ETHEREUM_HWI
+
+    def __init__(
+        self,
+        private_key_path: Optional[str] = None,
+        password: Optional[str] = None,
+        extra_entropy: Union[str, bytes, int] = "",
+        **kwargs: Any,
+    ) -> None:
+        """
+        Instantiate an ethereum crypto object.
+
+        :param private_key_path: the private key path of the agent
+        :param password: the password to encrypt/decrypt the private key.
+        :param extra_entropy: add extra randomness to whatever randomness your OS can provide
+        :param kwargs: extra keyword arguments
+        """
+        self._device_index = kwargs.pop(
+            "default_device_index",
+            DEFAULT_DEVICE_INDEX,
+        )
+        self._keypair_index = kwargs.pop(
+            "default_keypair_index",
+            DEFAULT_KEYPAIR_INDEX,
+        )
+        account = HWIAccount(
+            default_device=self._device_index,
+            default_key_index=self._keypair_index,
+        )
+        super().__init__(
+            private_key_path=private_key_path,
+            password=password,
+            extra_entropy=extra_entropy,
+            entity=account,
+        )
+
+    @property
+    def private_key(self) -> str:
+        """
+        Return a private key.
+
+        64 random hex characters (i.e. 32 bytes) + "0x" prefix.
+        """
+
+        raise NotImplementedError()
+
+    @property
+    def public_key(self) -> str:
+        """
+        Return a public key in hex format.
+
+        128 hex characters (i.e. 64 bytes) + "0x" prefix.
+
+        :return: a public key string in hex format
+        """
+        return self.entity.public_key
```

Comment:
No `0x`? Just checking because below you manually add it

Reply:
No `0x` is required